processor/metricstransformprocessor/metrics_transform_processor_test.go

```diff
@@ -239,54 +65,92 @@ func TestMetricsTransformProcessor(t *testing.T) {
 			require.Equal(t, 1, len(got))
 			gotMD := pdatautil.MetricsToMetricsData(got[0])
 			require.Equal(t, 1, len(gotMD))
-			require.Equal(t, len(test.outMN), len(gotMD[0].Metrics))
+			actualOut := gotMD[0].Metrics
+			require.Equal(t, len(test.out), len(actualOut))
 
-			targetNameToTransform := make(map[string]Transform, len(test.transforms))
-			for _, transform := range test.transforms {
-				targetName := transform.MetricName
-				if transform.NewName != "" {
-					targetName = transform.NewName
-				}
-				targetNameToTransform[targetName] = transform
-			}
-			for idx, out := range gotMD[0].Metrics {
-				// check name
-				assert.Equal(t, test.outMN[idx], out.MetricDescriptor.Name)
-				// check labels
-				// check the updated or inserted is correctly updated
-				transform, ok := targetNameToTransform[out.MetricDescriptor.Name]
-				if ok {
-					for lidx, l := range out.MetricDescriptor.LabelKeys {
-						assert.Equal(t, test.outLabels[lidx], l.Key)
-					}
-				}
-				// check the original is untouched if insert
-				if transform.Action == Insert && out.MetricDescriptor.Name == transform.MetricName {
-					for lidx, l := range out.MetricDescriptor.LabelKeys {
-						assert.Equal(t, test.inLabels[lidx], l.Key)
-					}
-				}
+			for idx, out := range test.out {
+				assert.Equal(t, *out, *actualOut[idx])
 			}
+
 			assert.NoError(t, mtp.Shutdown(ctx))
 		})
 	}
 }
 
+func TestComputeDistVals(t *testing.T) {
```

Comment:
👌

Reply:
This is good but maybe pull the exemplar selection out of the same function so its properly isolated for testing.

Reply:
Added exemplar test. Please let me know if this is a valid test for randomness.

Reply:
I have isolated the exemplar picker and changed the test. This is very useful feedback, very good to know. Thanks!